cdap-docs/developers-manual/source/building-blocks/namespaces.rst

```diff
@@ -102,7 +102,15 @@ Namespace Examples
 - The CDAP :ref:`Command Line Interface <cli>` is namespace-aware. You set the
   namespace you are currently using; the command prompt displays it as a visual reminder.
 
+.. _namespaces-custom-mapping:
 
-<Update this entire document with namespace mapping features>
-
-<Currently this seems to be the place to document impersonation too>
+Custom Mapping of Storage Providers
+===================================
+When creating a namespace, the underlying storage provider can also be configured per
+namespace. For example, a custom HBase namespace, Hive database, or HDFS directory can be
+specified to be used for data in a particular namespace.
+See :ref:`Namespace Configurations <http-restful-api-namespace-configs>`.
+
+When configuring these, CDAP will not manage the lifecycle of these entities; they must exist
```

Comment:
Do we need to say that we do delete the data inside these namespaces/directories ?